test/e2e/cluster_logging_es.go

```diff
@@ -121,14 +122,22 @@ func checkElasticsearchReadiness(f *framework.Framework) error {
 			framework.Logf("After %v failed to get services proxy request: %v", time.Since(start), errProxy)
 			continue
 		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), framework.SingleCallTimeout)
+		defer cancel()
+
 		// Query against the root URL for Elasticsearch.
 		response := proxyRequest.Namespace(api.NamespaceSystem).
+			Context(ctx).
 			Name("elasticsearch-logging").
 			Do()
 		err = response.Error()
 		response.StatusCode(&statusCode)
 
 		if err != nil {
+			if ctx.Err() != nil {
+				framework.Failf("After %v proxy call to elasticsearch-loigging failed: %v", time.Since(start), ctx.Err())
```

Comment:
https://github.com/kubernetes/kubernetes/issues/38305#issuecomment-265542104

This is trying to fix tests from hanging forever on proxy calls. Instead of hanging and failing an entire suite, we timeout after 5 minutes and fail the test that hung. We want to notice this hangs so hopefully we can debug the underlying issue. That's why we fail tests if it hangs.

There are some places that don't fail the test at the call site, but that's because it would fail further up the call chain.